web/api/v1/api.go

```diff
@@ -255,7 +255,17 @@ func respond(w http.ResponseWriter, data interface{}) {
 
 func respondError(w http.ResponseWriter, apiErr *apiError, data interface{}) {
 	w.Header().Set("Content-Type", "application/json")
-	w.WriteHeader(422)
```

Comment:
Not returning a 422 at all doesn't seem correct.
`errorExec` happens if execution semantically failed but the input (query string, timestamps) was syntactically correct. This is 422.

`errorCanceled` currently only happens if the query engine is not accepting requests, this would be `503 Service Unavailable`. Arguable `errorTimeout` is the same, as it generally indicates overload.